wallet_test.go

```diff
@@ -0,0 +1,121 @@
+// Package implements a wallet, that holds bitcoin and operations: to deposit and withdraw money and get the wallet's balance
+
+package wallet
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestWallet_GetBalance(t *testing.T) {
+	tests := []struct {
+		name string
+		w    *Wallet
+		want Bitcoin
+	}{
+		{
+			name: "zero balance",
+			w:    &Wallet{balance: 0},
+			want: 0,
+		},
+		{
+			name: "positive balance",
+			w:    &Wallet{balance: 1},
+			want: 1,
+		},
+		{
+			name: "negative balance",
+			w:    &Wallet{balance: -1},
+			want: -1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.w.GetBalance(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Wallet.GetBalance() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestWallet_Withdraw(t *testing.T) {
+	type args struct {
+		b Bitcoin
+	}
+	tests := []struct {
+		name    string
+		w       *Wallet
+		args    args
+		wantErr bool
```

Comment:
In our tests we want to not only check that an error occurred, but also that it was the exact error we expected to occur. To help with that, we should use the fact that "errors are values" in Go. Define concrete error values and check for them specifically. Try to figure this out, ping me if you have any problems

https://go.dev/blog/errors-are-values